luigi_pipeline/seqr_gcnv_loading.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import sys
+
+import luigi
+import hail as hl
+
+from lib.model.gcnv_mt_schema import SeqrGCNVVariantSchema, SeqrGCNVGenotypesSchema, SeqrGCNVVariantsAndGenotypesSchema
+from luigi_pipeline.seqr_loading_optimized import SeqrVCFToVariantMTTask, BaseVCFToGenotypesMTTask, BaseMTToESOptimizedTask
+from sv_pipeline.genome.utils.mapping_gene_ids import load_gencode
+
+
+logger = logging.getLogger(__name__)
+
+SAMPLE_ID_REGEX = r'(.+)_v\d+_Exome_(C|RP-)\d+$'
+
+class SeqrGCNVVariantMTTask(SeqrVCFToVariantMTTask):
+    # Overrided inherited required params.
+    reference_ht_path = ""
+    clinvar_ht_path = ""
+    sample_type = "WES"
+    genome_version = "38"
+    dont_validate = True
+    dataset_type = "SV"
+
+    RUN_VEP = False
+    SCHEMA_CLASS = SeqrGCNVVariantSchema
+
+    def annotate_old_and_split_multi_hts(self, mt, *args, **kwargs):
+        return mt
+
+    def add_37_coordinates(self, mt, *args, **kwargs):
+        # TODO.. implement this (it was out of scope in the port but we want it eventually)
+        return mt
+
+    def generate_callstats(self, mt, *args, **kwargs):
+        return mt
+
+    def import_dataset(self):
+        ht = hl.import_table(self.source_paths[0], impute=True, min_partitions=500)
+        mt = ht.to_matrix_table(
+            row_key=['variant_name', 'svtype'], col_key=['sample_fix'],
+            # Analagous to CORE_COLUMNS = [CHR_COL, SC_COL, SF_COL, CALL_COL, IN_SILICO_COL] in the old implementation
+            row_fields=['chr', 'vac', 'vaf', 'strvctvre_score'],
+        )
+
+        # rename the sample id column before the sample subset happens
+        mt = mt.transmute_cols(s = mt.sample_fix.first_match_in(SAMPLE_ID_REGEX)[0])
+
+        # This rename helps disambiguate between the 'start' & 'end' that are aggregations
+        # over samples and the start and end of each sample.
+        return mt.rename({'start': 'sample_start', 'end': 'sample_end'})
```

Comment:
I think `start` and `end` are for variants, not samples. But I'm unsure about gCNV.

Reply:
👍 

yep!  I think the situation here is we're getting a tsv of values where each row corresponds to the `start` and `end` of a variant & sample combination.  We then aggregate over the samples to get the variant `start` and `end` when we do the row annotations:

```
    @row_annotation()
    def start(self):
        return hl.agg.min(self.mt.sample_start)

    @row_annotation()
    def end(self):
        return hl.agg.max(self.mt.sample_end)
```

I was initially fine leaving the `start` and `end` named what they were, but hail was very unhappy with something like this:

```
    @row_annotation()
    def start(self):
        return hl.agg.min(self.mt.start)

    @row_annotation()
    def end(self):
        return hl.agg.max(self.mt.end)
```
because the field names already exist.